viewing_party/party.py

```diff
@@ -1,23 +1,217 @@
+         average = 0.0
+         return average
+    else:
+        for  value in user_data.values():   
+            ratings =[]
+            rate_list= value[0]
+            ratings.append(rate_list["rating"])
+            rate_list= value[1]
+            ratings.append(rate_list["rating"])
+            rate_list= value[2]
+            ratings.append(rate_list["rating"])
+            rate_list= value[3]
+            ratings.append(rate_list["rating"])
+            rate_list= value[4]
+            ratings.append(rate_list["rating"])
+            rate_list= value[5]
+            ratings.append(rate_list["rating"])
+            average = sum(ratings)/ len(ratings)
+            return average
+
+def get_most_watched_genre(user_data):
+    genre_count = {}
+    if user_data["watched"] == []:
+        most_watched = None
+        return most_watched
+        
+    else:
+        for movie in user_data["watched"]:
+            if movie["genre"] not in genre_count.keys():
+                genre_count[movie["genre"]] = 1
+            elif movie["genre"] in genre_count.keys():
+                genre_count[movie["genre"]] += 1
+        most_watched = max(genre_count, key=genre_count.get)
+        return most_watched
+
+
+
+ 
 
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
 # -----------------------------------------
 
+user_watched= [FANTASY_1,
+                FANTASY_2,
+                FANTASY_3,
+                ACTION_1,
+                INTRIGUE_1,
+                INTRIGUE_2,
+]
+
+first_friend_watched_data = [
+                FANTASY_1,
+                FANTASY_3,
+                FANTASY_4,
+                HORROR_1
+               
+            ]
+second_friend_watched_data = [ FANTASY_1,
+                ACTION_1,
+                INTRIGUE_1,
+                INTRIGUE_3,
+                ]
+def get_unique_watched(user_data):
+    unique_list = [] 
+    comparison_1 =[]
+    comparison_2 =[]
+    for movie in user_data["watched"]:
+        if movie not in first_friend_watched_data:
+            comparison_1.append(movie)
+    for movie in user_data["watched"]:
+        if movie  not in second_friend_watched_data:
+            comparison_2.append(movie)
+    for movie in comparison_2:
+        if movie in comparison_1 and comparison_2:
+           unique_list.append(movie)
+    return unique_list
+    
+
+def get_friends_unique_watched(user_data):
+    friends_unique_movies = []
+    for friend in user_data["friends"]:
+        for movie in friend["watched"]:
+            if movie not in user_data["watched"]:
+                friends_unique_movies.append(movie)
+    
+    return friends_unique_movies 
+  
         
 # -----------------------------------------
 # ------------- WAVE 4 --------------------
 # -----------------------------------------
 
+
+user_watched = []
+friends_watched = []
+comparison_list =[]
+recommendations = []
+def get_available_recs(user_data):

```

Comment:
think about the function `get_friends_unique_watched()`. How could it be used in this function?